modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/resolver/ValueResolverFactoryTypeVisitor.java

```diff
@@ -69,6 +70,15 @@ public ValueResolverFactoryTypeVisitor(DslSyntaxResolver dslSyntaxResolver, Stri
     this.dslSyntaxResolver = dslSyntaxResolver;
     this.defaultValue = defaultValue;
     this.acceptsReferences = acceptsReferences;
+    initializeValueResolverDelegates();
+  }
+
+  private void initializeValueResolverDelegates() {
+    valueResolverDelegates.put(metadataType -> MediaType.class.equals(getType(metadataType)),
```

Comment:
why is this implemented like this. Sounds super overkill. Specially the DataType part.

Reply:
Before, this class was in the extensions-spring-support module, and relied on ValueResolverParsingDelegate like this one : https://github.com/mulesoft/mule/blob/master/modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/object/CharsetValueResolverParsingDelegate.java . 

The ValueResolverFactoryTypeVisitor only used 2, which only made use of the MetadataType and the value given, so I turned those 2 into the valueResolverDelegates map. The predicates are the implementation the accept method had, and the given value resolver is the same as the parse method would returned.

Reply:
I understand that. but this implementation is still criptic and inneficient. For instance, in the original implementation the map had several entries, but no real need to keep using a map if it's only two now.